src/ledger/LedgerManagerImpl.h

```diff
@@ -57,6 +57,11 @@ class LedgerManagerImpl : public LedgerManager
     medida::Counter& mLedgerAge;
     medida::Timer& mMetaStreamWriteTime;
     VirtualClock::time_point mLastClose;
+    bool mRebuildInMemoryState{false};
+    // References to buckets being applied during state rebuild; this is needed
+    // to ensure we do not accidentally garbage-collect buckets during in-memory
+    // state rebuild
+    std::set<std::shared_ptr<Bucket>> mRetainBucketsDuringRebuild;
```

Comment:
This seems a bit odd - shouldn't that be referenced by catchup as it's the "thing" that applies buckets: we just need to make sure that buckets are referenced as soon as we create the catchupwork that rebuilds state?

Reply:
Updated, bucket references are now passed to catchup (but collected prior to application start, which invokes garbage collection)